esmvalcore/_data_finder.py

```diff
@@ -231,65 +284,104 @@ def _find_input_dirs(variable, rootpath, drs):
+    short_name = variable["short_name"]
+    variable["short_name"] = variable["original_short_name"]
     input_dirs = _find_input_dirs(variable, rootpath, drs)
     filenames_glob = _get_filenames_glob(variable, drs)
     files = find_files(input_dirs, filenames_glob)
-    variable['short_name'] = short_name
+    variable["short_name"] = short_name
     return (files, input_dirs, filenames_glob)
 
 
 def get_input_filelist(variable, rootpath, drs):
     """Return the full path to input files."""
     # change ensemble to fixed r0i0p0 for fx variables
     # this is needed and is not a duplicate effort
-    if variable['project'] == 'CMIP5' and variable['frequency'] == 'fx':
+    if all([
+            variable['project'] == 'CMIP5', variable['frequency'] == 'fx',
+            variable.get('ensemble') != '*'
+    ]):
         variable['ensemble'] = 'r0i0p0'
     (files, dirnames, filenames) = _find_input_files(variable, rootpath, drs)
+
     # do time gating only for non-fx variables
-    if variable['frequency'] != 'fx':
-        files = select_files(files, variable['start_year'],
-                             variable['end_year'])
+    if variable["frequency"] != "fx":
+        files = select_files(files, variable["start_year"],
+                             variable["end_year"])
     return (files, dirnames, filenames)
 
 
 def get_output_file(variable, preproc_dir):
     """Return the full path to the output (preprocessed) file."""
-    cfg = get_project_config(variable['project'])
+    cfg = get_project_config(variable["project"])
 
     # Join different experiment names
-    if isinstance(variable.get('exp'), (list, tuple)):
+    if isinstance(variable.get("exp"), (list, tuple)):
         variable = dict(variable)
-        variable['exp'] = '-'.join(variable['exp'])
+        variable["exp"] = "-".join(variable["exp"])
 
     outfile = os.path.join(
         preproc_dir,
-        variable['diagnostic'],
-        variable['variable_group'],
-        _replace_tags(cfg['output_file'], variable)[0],
+        variable["diagnostic"],
+        variable["variable_group"],
+        _replace_tags(cfg["output_file"], variable)[0],
     )
-    if variable['frequency'] != 'fx':
-        outfile += '_{start_year}-{end_year}'.format(**variable)
-    outfile += '.nc'
+    if variable["frequency"] != "fx":
+        outfile += "_{start_year}-{end_year}".format(**variable)
+    outfile += ".nc"
     return outfile
 
 
 def get_statistic_output_file(variable, preproc_dir):
     """Get multi model statistic filename depending on settings."""
     template = os.path.join(
         preproc_dir,
-        '{diagnostic}',
-        '{variable_group}',
-        '{dataset}_{mip}_{short_name}_{start_year}-{end_year}.nc',
+        "{diagnostic}",
+        "{variable_group}",
+        "{dataset}_{mip}_{short_name}_{start_year}-{end_year}.nc",
     )
 
     outfile = template.format(**variable)
 
     return outfile
+
+
+def dir_to_var(dirname, basepath, project, drs):
```

Comment:
this is a public function, so you either add numpy-style API documentation for it, or you make it private (even if private I'd reckon it needs a bit more description - lots of operations it performs and not well documented)